extensions/binding/org.eclipse.smarthome.binding.digitalstrom/src/main/java/org/eclipse/smarthome/binding/digitalstrom/handler/DeviceHandler.java

```diff
@@ -0,0 +1,928 @@
+import org.eclipse.smarthome.binding.digitalstrom.internal.providers.DsThingTypeProvider;
+import org.eclipse.smarthome.config.core.Configuration;
+import org.eclipse.smarthome.core.library.types.DecimalType;
+import org.eclipse.smarthome.core.library.types.IncreaseDecreaseType;
+import org.eclipse.smarthome.core.library.types.OnOffType;
+import org.eclipse.smarthome.core.library.types.PercentType;
+import org.eclipse.smarthome.core.library.types.StopMoveType;
+import org.eclipse.smarthome.core.library.types.StringType;
+import org.eclipse.smarthome.core.library.types.UpDownType;
+import org.eclipse.smarthome.core.thing.Bridge;
+import org.eclipse.smarthome.core.thing.Channel;
+import org.eclipse.smarthome.core.thing.ChannelUID;
+import org.eclipse.smarthome.core.thing.Thing;
+import org.eclipse.smarthome.core.thing.ThingStatus;
+import org.eclipse.smarthome.core.thing.ThingStatusDetail;
+import org.eclipse.smarthome.core.thing.ThingStatusInfo;
+import org.eclipse.smarthome.core.thing.ThingTypeUID;
+import org.eclipse.smarthome.core.thing.binding.BaseThingHandler;
+import org.eclipse.smarthome.core.thing.binding.ThingHandler;
+import org.eclipse.smarthome.core.thing.binding.builder.ChannelBuilder;
+import org.eclipse.smarthome.core.thing.binding.builder.ThingBuilder;
+import org.eclipse.smarthome.core.types.Command;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.google.common.collect.Sets;
+
+/**
+ * The {@link DeviceHandler} is responsible to handling the configuration, load supported channels of an
+ * digitalSTROM device and handling commands which are send to one of the channels. <br>
+ * <br>
+ * For that it uses the {@link BridgeHandler} to execute the actual command and implements the
+ * {@link DeviceStatusListener} to get informed about changes from the accompanying {@link Device}.
+ *
+ * @author Michael Ochel - Initial contribution
+ * @author Matthias Siegele - Initial contribution
+ *
+ */
+public class DeviceHandler extends BaseThingHandler implements DeviceStatusListener {
+
+    private Logger logger = LoggerFactory.getLogger(DeviceHandler.class);
+
+    /**
+     * The {@link DsThingTypeProvider} add the supported thing types
+     */
+    public final static Set<ThingTypeUID> SUPPORTED_THING_TYPES = Sets.newHashSet();
+
+    private String dSID = null;
+
+    private Device device;
+
+    private BridgeHandler dssBridgeHandler;
+
+    private Command lastComand = null;
+
+    private String currentChannel = null;
+
+    public DeviceHandler(Thing thing) {
+        super(thing);
+    }
+
+    @Override
+    public void initialize() {
+        logger.debug("Initializing DigitalSTROM Device handler.");
+        String configdSID = getConfig().get(DigitalSTROMBindingConstants.DEVICE_DSID).toString();
+        if (!configdSID.isEmpty()) {
+            dSID = configdSID;
+            if (getDssBridgeHandler() == null) {
```

Comment:
remove this "if", only keep the "else" - you must not set yourself HANDLER_MISSING_ERRORs, this is done by the framework.